core/src/main/java/com/sequenceiq/cloudbreak/service/stack/StackUpgradeService.java

```diff
@@ -1,27 +1,43 @@
 package com.sequenceiq.cloudbreak.service.stack;
 
+import static com.sequenceiq.cloudbreak.cloud.aws.common.AwsConstants.AwsVariant.AWS_NATIVE_VARIANT;
+import static com.sequenceiq.cloudbreak.cloud.aws.common.AwsConstants.AwsVariant.AWS_VARIANT;
+
 import javax.inject.Inject;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import org.springframework.stereotype.Component;
 
 import com.sequenceiq.cloudbreak.auth.altus.EntitlementService;
 import com.sequenceiq.cloudbreak.auth.crn.Crn;
-import com.sequenceiq.cloudbreak.cloud.aws.common.AwsConstants;
 import com.sequenceiq.cloudbreak.view.StackView;
 
 @Component
 public class StackUpgradeService {
 
+    private static final Logger LOGGER = LoggerFactory.getLogger(StackUpgradeService.class);
+
     @Inject
     private EntitlementService entitlementService;
 
-    public String calculateUpgradeVariant(StackView stack, String userCrn) {
+    public String calculateUpgradeVariant(StackView stack, String userCrn, boolean keepVariant) {
         String variant = stack.getPlatformVariant();
-        String accountId = Crn.safeFromString(userCrn).getAccountId();
-        boolean migrationEnable = entitlementService.awsVariantMigrationEnable(accountId);
-        if (migrationEnable) {
-            if (AwsConstants.AwsVariant.AWS_VARIANT.variant().value().equals(variant)) {
-                variant = AwsConstants.AwsVariant.AWS_NATIVE_VARIANT.variant().value();
+        LOGGER.debug("About to calculate variant (current is: {})", variant);
+        if (keepVariant) {
+            LOGGER.debug("Keeping the original variant is requested, therefore the following one is going to be returned: {}", variant);
+        } else {
+            String accountId = Crn.safeFromString(userCrn).getAccountId();
+            if (AWS_VARIANT.variant().value().equals(variant)) {
+                if (entitlementService.awsVariantMigrationEnable(accountId)) {
+                    LOGGER.debug("Variant migration is enabled and the {} variant is detected, therefore the following one is going to return " +
+                            "to change the original one {}", AWS_VARIANT.variant().value(), AWS_NATIVE_VARIANT.variant().value());
+                    variant = AWS_NATIVE_VARIANT.variant().value();
+                } else {
+                    LOGGER.debug("AWS variant detected but the variant migration entitlement is not enabled for the account (id: {})", accountId);
+                }
+            } else {
+                LOGGER.debug("The following variant is going to be returned: {}", variant);
             }
```

Comment:
you can use 1 if statement here, there is no need to log information about entitlement enablement, since EntitlementService already does that, so this would be enough:
```
if (AWS_VARIANT.variant().value().equals(variant) && entitlementService.awsVariantMigrationEnable(accountId)) {
            LOGGER.debug("Variant migration is enabled and the {} variant is detected, therefore the following one is going to return " +
                    "to change the original one {}", AWS_VARIANT.variant().value(), AWS_NATIVE_VARIANT.variant().value());
            variant = AWS_NATIVE_VARIANT.variant().value();
        } else {
            LOGGER.debug("The following variant is going to be returned: {}", variant);
        }
```